pkg/controller/machine_bootstrap_token.go

```diff
@@ -56,41 +58,61 @@ func (c *controller) addBootstrapTokenToUserData(machineName string, driver driv
 	return nil
 }
 
-func (c *controller) getBootstrapTokenOrCreateIfNotExist(machineName string) (secret *v1.Secret, err error) {
-	tokenID, secretName := getTokenIDAndSecretName(machineName)
-
-	secret, err = c.targetCoreClient.CoreV1().Secrets(metav1.NamespaceSystem).Get(secretName, metav1.GetOptions{})
+func (c *controller) getBootstrapTokenOrCreateIfNotExist(machineLabels map[string]string, machineName string) (secret *v1.Secret, err error) {
+	value, ok := machineLabels["instance-group"]
+	if !ok || value == "" {
+		return nil, fmt.Errorf("missing instance-group label from machine: '%v'", machineName)
+	}
+	octets := strings.Split(value, "-")
+	if len(octets) == 0 {
+		return nil, fmt.Errorf("failed to get nodegroup name from machine: '%v'", machineName)
+	}
+	labelSelector := fmt.Sprintf("node-manager.deckhouse.io/node-group=%s", octets[0])
```

Comment:
label `instance-group` contains node group + zone. In the node group we can have dash symbol, for example `worker-big`
![image](https://github.com/user-attachments/assets/50b1931b-bf0a-41ae-8a3e-848e5157ee49)
![image](https://github.com/user-attachments/assets/8bb62ed5-7c43-4dee-b612-b548b4210cfa)

in this case we will get `worker` instead of `worker-big`. Also zone also can have dash symbol, for yandex, for example. Working with split function will be hard to extract zone name.

I suggest get ng name from spec in the machine. from nodeTemplate
![image](https://github.com/user-attachments/assets/069708ce-57a1-40ff-9ced-b40e443ae69a)